internal/querynodev2/delegator/delegator.go

```diff
@@ -700,12 +767,15 @@ func NewShardDelegator(ctx context.Context, collectionID UniqueID, replicaID Uni
 		loader:          loader,
 		factory:         factory,
 		queryHook:       queryHook,
+		chunkManager:    chunkManager,
+		partitionStats:  make(map[UniqueID]*storage.PartitionStats),
 	}
 	m := sync.Mutex{}
 	sd.tsCond = sync.NewCond(&m)
 	if sd.lifetime.Add(lifetime.NotStopped) == nil {
 		go sd.watchTSafe()
 	}
 	log.Info("finish build new shardDelegator")
+	sd.maybeReloadPartitionStats(ctx)
```

Comment:
Not sure how long it will take. Perhaps we can optimize it with goroutine in the future?

Reply:
True, I really do not know how long it will take to load the file from s3